homeassistant/components/pure_energie/config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Config flow for Pure Energie integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from gridnet import Device, GridNet, GridNetConnectionError
+import voluptuous as vol
+
+from homeassistant.components import zeroconf
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+
+class PureEnergieFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Config flow for Pure Energie integration."""
+
+    VERSION = 1
+    discovered_host: str
+    discovered_device: Device
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
+        errors = {}
+
+        if user_input is not None:
+            try:
+                device = await self._async_get_device(user_input[CONF_HOST])
+            except GridNetConnectionError:
+                errors["base"] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(device.n2g_id)
```

Comment:
Please add `raise_on_progress=False` here since there may be a discovery flow in progress and the user flow should take precedence.